site/concurrency-primitives.md

```diff
@@ -6,7 +6,61 @@ In the [`fs2.concurrent` package](https://github.com/functional-streams-for-scal
 - `Topic[F, A]`
 - `Signal[F, A]`
 
-These data structures could be very handy in a few cases. See below examples of each of them originally taken from [gvolpe examples](https://github.com/gvolpe/advanced-http4s/tree/master/src/main/scala/com/github/gvolpe/fs2):
+## Simple Examples
+
+### Topic
+
+(based on [Pera Villega](https://perevillega.com/)'s example [here](https://underscore.io/blog/posts/2018/03/20/fs2.html))
+
+Topic implements the publish-subscribe pattern. In the following example, `publisher` and `subscriber` are started concurrently with the publisher continuously publishing the string `"1"` to the topic and the subscriber consuming it until it received four elements, including the initial element, the string `"Topic start"`.
+
+```scala
```

Comment:
If you change this to `scala mdoc`, the example will get compiled/tested as part of site generation. In this case, it'll show that the example is using the old CE2 / FS2 2.x API and hence needs to be updated for CE3 / FS2 3.x.